clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcsServerClusterProvider.java

```diff
@@ -480,6 +485,15 @@ public Map<String, Set<EcsServerCluster>> getClusters() {
     return clusterMap;
   }
 
+  public Map<String, Set<EcsServerCluster>> getClusters0(String application) {
```

Comment:
getClusters0 seems like a bad function name.  I'm... wondering why this method vs. the getClusterSummaries up above.

Reply:
Im following the same pattern/naming convention established here for the AWS provider https://github.com/spinnaker/clouddriver/blob/0226100a95b16e1176a296198b1f892c4514e4d6/clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/view/AmazonClusterProvider.groovy#L70:L78 

But for the ECS provider the Cluster view never implemented the clusterSummaries. It was always returning the clusterDetails. 

Which is another area of improvement. getClusterSummaries should return only the names of the serverGroups and loadbalancers. getClusterDetails should return the full details of the serverGroups and loadbalancers. 

Will actually address this in this PR which makes sense to improve here.

Reply:
Key differences:
* getClusters0 in AWS is a parameterized private method which takes an argument to control whether details are added. In this case, there's no difference between the getDetails & getSummaries calls (which is a question - WHY there isn't a difference, but not looked at the data returned).  
* As such, it's extra code that doesn't help code deduplication.  AN alternative is making getClusters0 private but I'd prefer merge the logic and just have one call the other vs. a separate method that's private that both call increasing the call stack.

Reply:
Sounds good! No objection in implementing the getDetails and getSummaries better :) Just pushed the changes + i verified that the API responses before and after the change are the same.

With the difference that previously getClusterSummaries was returning All the ECS clusters/services for all the accounts (no application filtered). After the change it will return only the application clusters/services. 

getSummaries is called via the applicationController -> GET /applications/<applicationName>

getDetails is called via the serverGroup controller -> GET /applications/<applicationName>/serverGroups